common/src/main/java/com/datastrato/gravitino/dto/util/DTOConverters.java

```diff
@@ -316,13 +316,19 @@ public static Partitioning toDTO(Transform transform) {
 
     } else if (transform instanceof Transforms.ListTransform) {
       Transforms.ListTransform listTransform = (Transforms.ListTransform) transform;
-      return ListPartitioningDTO.of(
-          listTransform.fieldNames(), (ListPartitionDTO[]) toDTOs(listTransform.assignments()));
+      ListPartitionDTO[] assignments =
+          Arrays.stream(listTransform.assignments())
+              .map(DTOConverters::toDTO)
+              .toArray(ListPartitionDTO[]::new);
+      return ListPartitioningDTO.of(listTransform.fieldNames(), assignments);
```

Comment:
What would be the problem if we use the way we forced the conversion before the change? Can we add tests to cover this change?